src/NumericalAlgorithms/RootFinding/GslMultiRootFunctionFdf.hpp

```diff
@@ -0,0 +1,32 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+// \file
+// Declares classes useful for interfacing with GSL MultiRoot functions.
+
+#pragma once
+
+#include <gsl/gsl_matrix.h>
+#include <gsl/gsl_vector.h>
+
+// Base class for providing functions to GSL MultiRootFinder
+template <size_t Dim>
+class GslMultiRootFunctionFdf {
+ public:
+  virtual ~GslMultiRootFunctionFdf() = default;
+  virtual std::array<double, Dim> f(const std::array<double, Dim>& x) const = 0;
+  virtual std::array<std::array<double, Dim>, Dim> df(
+      const std::array<double, Dim>& x) const = 0;
+};
+
+// Internal use functions for interfacing with GSL
+template <size_t Dim>
+int Gsl_MultiRootFunctionFdf_Wrapper_f(const gsl_vector* v, void* params,
```

Comment:
Since these are only for internal use by the root finder, should we just put them in that file?  Will they be used by something else in the future?  (Also naming.)

Reply:
With this and once the base class is gotten rid of I think we'll be able to get rid of the GslMultiRootFunctionFdf files entirely